src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        'The side of the figure cannot be less than or equal to zero',
+      );
```

Comment:
[CODE STYLE] - When throwing an error, try to provide a detailed description in the error message. In your case, it would be more helpful to specify which side of the figure is less than or equal to zero.